### ISSUE TITLE
[Question] Modify env configuration as a function of training progress

### ISSUE THREAD

**Issue by npit** [DLR-RM/stable-baselines3#1686]
### ❓ Question

Similarly to modifying the learning rate as a function of training progress, I want to modify some environment parameter as a function of training time remaining, specifically hyperparameters controlling episode termination.

I can do this in a hacky way by directly accessing and modifying the envs in a callback, using the [provided interface of vectorized envs](https://stable-baselines3.readthedocs.io/en/master/guide/vec_envs.html).

Is there an alternative recommended way to do this instead? Do you have any other suggestions / observations?

Thanks

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> I can do this in a hacky way by directly accessing and modifying the envs in a callback, using the [provided interface of vectorized envs](https://stable-baselines3.readthedocs.io/en/master/guide/vec_envs.html).

Callbacks have access to the env via `self.training_env` and this is the recommended way to go, I'm not sure why you consider that to be hacky.
